src/coord/src/coord.rs

```diff
@@ -4096,6 +4111,121 @@ impl Coordinator {
                 .expect("sending to internal_cmd_tx cannot fail");
         });
     }
+
+    /// Validate that sources which only support a single materialization are only materialized once
+    fn validate_single_materializations(
+        &self,
+        index_to_build: Option<GlobalId>,
+        index_view: View,
+        expr: OptimizedMirRelationExpr,
+    ) -> Result<(), CoordError> {
+        let id = if let Some(id) = index_to_build {
```

Comment:
if this function does nothing when `index_to_build` is `None` it probably better to make the parameter a `GlobalId` and let the caller check it since it's a required parameter